File: backend/app/models/project.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Integer, Text, DateTime
from app.db.session import Base

def utc_now() -> datetime:
    return datetime.now(timezone.utc)

class Project(Base):
    __tablename__ = "projects"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    name = Column(String(255), nullable=False, index=True)
    description = Column(Text, nullable=True)
    tempo = Column(Integer, nullable=False, default=120)
    key = Column(String(10), nullable=False, default="C")
    time_signature = Column(String(10), nullable=False, default="4/4")
    created_at = Column(DateTime(timezone=True), nullable=False, default=utc_now)
    updated_at = Column(DateTime(timezone=True), nullable=False, default=utc_now, onupdate=utc_now)

File: frontend/src/types/project.ts
export interface Project {
  id: string;
  name: string;
  description?: string | null;
  tempo: number;
  key: string;
  time_signature: string;
  created_at: string;
  updated_at: string;
}

export interface CreateProjectDTO {
  name: string;
  description?: string;
  tempo?: number;
  key?: string;
  time_signature?: string;
}

export interface UpdateProjectDTO {
  name?: string;
  description?: string;
  tempo?: number;
  key?: string;
  time_signature?: string;
}

export interface HealthCheckResponse {
  status: string;
  service: string;
}
